lib/personalized_search/product_preferences_manager.dart

```diff
@@ -13,7 +12,6 @@ class ProductPreferencesManager {
   final ProductPreferencesSelection _productPreferencesSelection;
   AvailableProductPreferences? _availableProductPreferences;
 
-  @protected
```

Comment:
Did you consider keeping `@protected` and adding `@visibleForTesting` instead?

Reply:
@peterwvj I did not, because I didn't know `@visibleForTesting` :)
That being said, the fact that I removed `@protected` has nothing to do with the fact that it's a unit test.
In the real world, there's no reason to force developers to create a class that overrides `ProductPreferencesManager` - which is mandatory if we keep the `@protected` annotation.
For the record, initially I added the `@protected` in the smoothie code that @stephanegigandet moved here. I can't remember the good idea behind it - except the joy of using a Java keyword that I thought did not exist in dart.